core/src/main/java/hudson/security/AccessControlled.java

```diff
@@ -61,6 +65,9 @@ default void checkAnyPermission(@NonNull Permission... permission) throws Access
      * Convenient short-cut for {@code getACL().hasPermission(permission)}
      */
     default boolean hasPermission(@NonNull Permission permission) {
+        if (Jenkins.getAuthentication2().equals(ACL.SYSTEM2)) {
+            return true;
+        }
```

Comment:
https://github.com/jenkinsci/jenkins/blob/9116ce84feb81d6c709ec425dde131baacf543c8/core/src/main/java/hudson/security/AccessControlled.java#L89-L91 was already enforcing this logic.